### ISSUE TITLE
Multiple Terminals for Reverse Shell and interactive terminals

### ISSUE THREAD

**Issue by isamu-isozaki** [ipa-lab/hackingBuddyGPT#90]
Hi! My understanding is that hackingbuddygpt is optimized for privilege escalation so didn't need to integrate this but I think if we are going for autopentesting we need multiple terminal support for say reverse shell and interactive terminals for like su(the ssh may not be enough since sometimes we want to change user to a user which cannot be connected via ssh). I think [auto-pen-bench](https://github.com/lucagioacchini/auto-pen-bench) already has some starter code to enable this

**Comment by andreashappe** [comment]
hi @isamu-isozaki , the origins of hackingbuddygpt were in priv-esc, but we've been expanding in other areas so this is very relevant to us (;

**Comment by isamu-isozaki** [comment]
@andreashappe makes sense! I think for this, we might need to switch/add option for paramiko/pexpect since at least in fabric I don't think it can handle the interactive terminal part. For multiple terminals, honestly I'm not sure the best way to handle it in terms of input to llms.
For now, I plan to just play around with auto-pen-bench after adding vagrant to it and see how they might handle multi terminal

**Comment by andreashappe** [comment]
slowly but finally.. support for interactive terminals should now be better.. basically I'm working on enabling better SLM support for hackingbuddygpt. Next step will be unification of the web prototypes.
